src/Animations/babylon.animation.ts

```diff
@@ -79,7 +90,26 @@
             this.loopMode = loopMode === undefined ? Animation.ANIMATIONLOOPMODE_CYCLE : loopMode;
         }
 
-        // Methods   
+        // Methods
+        /**
+         * Add an event to this animation.
+         */
+        public addEvent(event: AnimationEvent) : void {
+            this._events.push(event);
+        }
+
+        /**
+         * Remove all events found at the given frame
+         * @param frame
+         */
+        public removeEvents(frame:number) : void {
+            for (var index = 0; index < this._events.length; index++) {
+                if (this._events[index].frame === frame) {
+                    this._events.splice(index, 1);
```

Comment:
Sorry, should have noticed before - this will skip the element after the splice. You will need to deduct 1 from the index (or use splice(index--,1)), otherwise some elements won't be iterated...

Leave this open, just push a change to this branch and it will appear here.